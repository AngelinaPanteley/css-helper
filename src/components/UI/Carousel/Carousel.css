.carousel .thumbs-wrapper {
  display: none;
}

.carousel .slide {
  height: calc(100vh - 79px);
  position: relative;
  z-index: 1;
}

.carousel .slide > div {
  height: 100%;
}

.carousel .slide.selected {
  z-index: 5;
}

.carousel .slide img {
  width: auto;
  min-width: 100%;
  min-height: 100%;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%,-50%);
  object-fit: cover;
}

.carousel .slide .legend {
  background: transparent;
}

.carousel .control-dots {
  padding: 0;
}

.carousel .control-dots .dot {
  box-shadow: none;
  cursor: pointer;
}

.carousel .slider-wrapper.axis-horizontal .slider .slide:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(56,58,71,0.3);
  z-index: 1;
}

.carousel .carousel-status {
  display: none;
}

.carousel.carousel-slider .control-arrow:before {
  font-family: "css-helper";
  font-style: normal;
  font-weight: normal;
  speak: none;
  display: inline-block;
  text-decoration: inherit;
  width: 1em;
  margin-right: .2em;
  text-align: center;
  font-variant: normal;
  text-transform: none;
  line-height: 1em;
  margin-left: .2em;
  border: 0;
}

.carousel .control-next.control-arrow:before {
  content: '\e802';
  border: 0;
}

.carousel .control-prev.control-arrow:before {
  content: '\e801';
  border: 0;
}