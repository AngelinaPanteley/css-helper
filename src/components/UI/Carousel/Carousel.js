import React from "react";
import PropTypes from 'prop-types';
import { Carousel } from "react-responsive-carousel";
import "./lib/carousel.css";
import './Carousel.css';
import LinkTo from '../LintTo/LinkTo';

const carousel = (props) => {
  let slides;

  if (props.slides) {
    slides = props.slides.map((slide) => {
      return (
        <div key={slide.imageUrl}>
          <img src={slide.imageUrl} alt='Editor Background' />
          <LinkTo route={slide.route}
            onClick={props.onClick}>
            {slide.title}
          </LinkTo>
        </div>
      );
    });
  }

  return (
    <Carousel autoPlay infiniteLoop>
      {slides}
    </Carousel>
  );
};

carousel.propTypes = {
  slides: PropTypes.arrayOf(PropTypes.object),
  onClick: PropTypes.func.isRequired,
}

export default carousel;