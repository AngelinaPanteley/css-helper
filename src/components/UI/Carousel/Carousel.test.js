"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import Carousel from './Carousel.js';
import { MemoryRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe('>>>Carousel', () => {

  //SNAPSHOTS

  const props = {
    slides: [{
      imageUrl: 'image.png',
      route: '/route',
      title: 'title',
    }],
    onClick: () => { },
  }

  const carouselComponent = <MemoryRouter><Carousel {...props} /></MemoryRouter>;

  const component = renderer.create(carouselComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <Carousel {...props} />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders background image', () => {
    chaiExpect(wrapper.contains(
      <img src={props.slides[0].imageUrl} alt='Editor Background' />
    )).to.equal(true);
  });

})