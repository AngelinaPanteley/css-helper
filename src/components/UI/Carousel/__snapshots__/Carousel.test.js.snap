// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Carousel renders itself correctly 1`] = `
<div>
  <div
    className="carousel carousel-slider"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <button
      className="control-arrow control-prev control-disabled"
      onClick={[Function]}
      type="button"
    />
    <div
      className="slider-wrapper axis-horizontal"
      style={Object {}}
    >
      <ul
        className="slider animated"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "MozTransform": "translate3d(0%,0,0)",
            "MozTransitionDuration": "350ms",
            "MsTransform": "translate3d(0%,0,0)",
            "MsTransitionDuration": "350ms",
            "OTransform": "translate3d(0%,0,0)",
            "OTransitionDuration": "350ms",
            "WebkitTransform": "translate3d(0%,0,0)",
            "WebkitTransitionDuration": "350ms",
            "msTransform": "translate3d(0%,0,0)",
            "msTransitionDuration": "350ms",
            "transform": "translate3d(0%,0,0)",
            "transitionDuration": "350ms",
          }
        }
      >
        <li
          className="slide selected"
          onClick={[Function]}
        >
          <div>
            <img
              alt="Editor Background"
              src="image.png"
            />
            <a
              href="/route"
              onClick={[Function]}
            >
              title
            </a>
          </div>
        </li>
      </ul>
    </div>
    <button
      className="control-arrow control-next control-disabled"
      onClick={[Function]}
      type="button"
    />
    <ul
      className="control-dots"
    >
      <li
        className="dot selected"
        onClick={[Function]}
        value={0}
      />
    </ul>
    <p
      className="carousel-status"
    >
      1 of 1
    </p>
  </div>
  <div
    className="carousel"
  >
    <div
      className="thumbs-wrapper axis-vertical"
    >
      <button
        className="control-arrow control-prev control-disabled"
        onClick={[Function]}
        type="button"
      />
      <ul
        className="thumbs animated"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "MozTransform": "translate3d(0px,0,0)",
            "MozTransitionDuration": "350ms",
            "MsTransform": "translate3d(0px,0,0)",
            "MsTransitionDuration": "350ms",
            "OTransform": "translate3d(0px,0,0)",
            "OTransitionDuration": "350ms",
            "WebkitTransform": "translate3d(0px,0,0)",
            "WebkitTransitionDuration": "350ms",
            "msTransform": "translate3d(0px,0,0)",
            "msTransitionDuration": "350ms",
            "transform": "translate3d(0px,0,0)",
            "transitionDuration": "350ms",
          }
        }
      >
        <li
          className="thumb"
          onClick={[Function]}
        >
          <img
            alt="Editor Background"
            onLoad={[Function]}
            src="image.png"
          />
        </li>
      </ul>
      <button
        className="control-arrow control-next control-disabled"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</div>
`;
